lib/greenhouse_api/client.rb

```diff
@@ -33,18 +37,46 @@ def get_current_offer_for_application(application_id)
         http_method: :get,
         headers: headers,
         endpoint: "applications/#{application_id}/offers/current_offer",
-        params: {}
       )
+      compose_response(response)
+    end
 
-      if response.status == 200
-        Response.new(
-          body: response.body,
-          headers: response.headers,
-          status: response.status
-        )
-      else
-        response
-      end
+    sig { params(first_name: String, last_name: String, email: String, on_behalf_of_id: String, additional_args: String).returns(Response) }
+    def create_user(first_name:, last_name:, email:, on_behalf_of_id:, **additional_args)
+      body = { first_name: first_name, last_name: last_name, email: email }.merge(additional_args).to_json
+      response = request(
+        http_method: :post,
+        headers: headers.merge(ON_BEHALF_OF => on_behalf_of_id),
+        endpoint: "users",
+        body: body
+      )
+      compose_response(response)
+    end
+
+    sig { params(user: T::Hash[String, T.any(Integer, String)], on_behalf_of_id: String).returns(Response) }
```

Comment:
Had trouble figuring out how to specify the hash has to have at least one of the keys (`user_id`, `email`, or `employee_id`). If anyone knows, lmk!

Reply:
ohhh i'm not sure if you can specify the explicity strings like you could with typescript.. 🤔 
but if someone did, they'd get `400 => Invalid JSON Payload` right? i think that's probably fine